Firestore/core/src/firebase/firestore/immutable/sorted_map.h

```diff
@@ -195,6 +201,33 @@ class SortedMap : public impl::SortedMapBase {
     FIREBASE_UNREACHABLE();
   }
 
+  /**
+   * Returns an iterator pointing to the first entry in the map. If there are
+   * no entries in the map, begin() == end().
+   */
+  const_iterator begin() const {
```

Comment:
My comment about `begin`/`end` in `TreeSortedMap` applies here as well -- though I'm not suggesting to duplicate those. One additional interesting thing for the `SortedMap` iterator is that it's totally invalidated when the map switches from one underlying implementation to the other.